dvc/repo/experiments/remove.py

```diff
@@ -43,6 +44,26 @@ def remove(  # noqa: C901, PLR0912
 
     exp_ref_list: list[ExpRefInfo] = []
     queue_entry_list: list[QueueEntry] = []
+
+    if keep_selected:
+        # In keep_selected mode, identify all experiments and remove the unselected ones
+        all_exp_refs = exp_refs(repo.scm, git_remote)
+        selected_exp_names = resolve_selected_exp_names(
+            exp_names, git_remote, num, repo, rev
+        )
+
+        # Identify experiments to remove: all experiments - selected experiments
+        unselected_exp_refs = [
```

Comment:
why can't we go the regular path - collect experiments to remove and at the very end do the `_remove_commited_exps` ?

Reply:
At first I wanted to make sure my changes were super clear, but now that I refactored it, it looks much better like this and it stays clear.